solana/restaking/programs/restaking/src/lib.rs

```diff
@@ -94,18 +94,37 @@ pub mod restaking {
 
         // Call Guest chain program to update the stake if the chain is initialized
         if guest_chain_program_id.is_some() {
+            let service =
+                service.as_ref().ok_or(error!(ErrorCodes::MissingService))?;
+            let validator_key = match service {
+                Service::GuestChain { validator } => validator,
+            };
+            let borrowed_chain_data =
+                ctx.remaining_accounts[0].data.try_borrow_mut().unwrap();
```

Comment:
Why borrow mut if you’re only taking shared reference later on?
```suggestion
                ctx.remaining_accounts[0].data.try_borrow().unwrap();
```

Reply:
yes borrow is enough.